atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -51,51 +45,99 @@ class SendMailPlugin(ExitPlugin):
+        :param additional_addresses: list of str, always send a message to these email addresses
+        :param email_domain: str, email domain used when email addresses cannot be fetched via
+            kerberos principal
+        :param koji_hub: str, koji hub (xmlrpc)
+        :param koji_root: str, koji root (storage)
+        :param koji_proxyuser: str, proxy user
+        :param koji_ssl_certs_dir: str, path to "cert", "ca", and "serverca"
+        :param koji_krb_principal: str, name of Kerberos principal
+        :param koji_krb_keytab: str, Kerberos keytab
+        :param to_koji_submitter: bool, send a message to the koji submitter
+        :param to_koji_pkgowner: bool, send messages to koji package owners
         """
         super(SendMailPlugin, self).__init__(tasker, workflow)
-        self.send_on = send_on
+        self.send_on = list(send_on)
         self.url = url
-        self.submitter = submitter
-        self.pdc_url = pdc_url
-        self.pdc_verify_cert = pdc_verify_cert
-        self.pdc_component_df_label = pdc_component_df_label
-        self.pdc_secret_path = pdc_secret_path
-        self.pdc_contact_role = pdc_contact_role or self.PDC_CONTACT_ROLE
-        self.smtp_uri = smtp_uri
+        self.additional_addresses = additional_addresses
+        self.smtp_host = smtp_host
         self.from_address = from_address
         self.error_addresses = error_addresses
+        self.email_domain = email_domain
+        self.koji_hub = koji_hub
+        self.koji_root = koji_root
+        self.koji_auth_info = {
+            'proxyuser': koji_proxyuser,
+            'ssl_certs_dir': koji_ssl_certs_dir,
+            'krb_principal': koji_krb_principal,
+            'krb_keytab': koji_krb_keytab,
+        }
+        self.to_koji_submitter = to_koji_submitter
+        self.to_koji_pkgowner = to_koji_pkgowner
+        self.submitter = self.DEFAULT_SUBMITTER
+
+        try:
+            metadata = get_build_json().get("metadata", {})
+            self.koji_task_id = int(metadata['labels']['koji-task-id'])
+        except Exception:
+            self.log.exception("Failed to fetch koji task ID")
+            self.koji_task_id = None
+        else:
+            self.log.info("Koji task ID: %s" % self.koji_task_id)
 
-    def _should_send(self, rebuild, success, canceled):
+        try:
+            self.koji_build_id = int(self.workflow.exit_results.get(KojiPromotePlugin.key, {}))
+        except Exception:
+            self.log.exception("Failed to fetch koji build ID")
+            self.koji_build_id = None
+        else:
+            self.log.info("Koji build ID: %s" % self.koji_build_id)
+
+        try:
+            self.session = create_koji_session(self.koji_hub, self.koji_auth_info)
+        except Exception:
+            self.log.exception("Failed to connect to koji")
+            self.session = None
+        else:
+            self.log.info("Koji connection established")
```

Comment:
It is, later on it would be used in other functions to fetch details on koji task